common/models/todo.js

```diff
@@ -5,10 +5,11 @@ module.exports = function(Todo) {
 
   Todo.definition.properties.created.default = Date.now;
 
-  Todo.beforeSave = function(next, model) {
-    if (!model.id) model.id = 't-' + Math.floor(Math.random() * 10000).toString();
+  Todo.observe('before save', function(ctx, next) {
+    if (!ctx.Model.id)
+      ctx.Model.id = 't-' + Math.floor(Math.random() * 10000).toString();
```

Comment:
This hook is no longer needed, the id property is initialised by loopback to a new GUID.

I think the idea here was to use small semi-unique numbers that are easier to read than full GUIDs.

I am proposing to make a following change in the GUI to make IDs easier to read:
 - take the first 8 characters of the generated uid (the 4 low bytes of the timestamp)
 - use [proquint-](https://www.npmjs.com/package/proquint-) to convert these 8 characters into a readable string
 - show this readable representation in all place where we were rendering the original id (mostly conflict resolution views?)

Feel free to make UI changes in a new pull request. However, the hook should be removed as part of this patch.